pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractor.java

```diff
@@ -73,10 +73,12 @@ public GenericRow extract(GenericRecord from, GenericRow to) {
         to.putValue(fieldName, value);
       }
     } else {
+      Schema.Field field;
+      Object value;
       for (String fieldName : _fields) {
-        Object value = from.get(fieldName);
+        field = from.getSchema().getField(fieldName);
+        value = field == null ? null : from.get(field.pos());
```

Comment:
(nit) The declaration can happen inside
```suggestion
        Schema.Field field = from.getSchema().getField(fieldName);
        Object value = field == null ? null : from.get(field.pos());
```